main/src/com/google/refine/commands/expr/PreviewExpressionCommand.java

```diff
@@ -244,24 +277,35 @@ static protected void writeValue(StringBuffer sb, Object v, boolean quote) {
                 } else if (v instanceof HasFields) {
                     sb.append("[object " + v.getClass().getSimpleName() + "]");
                 } else if (v instanceof OffsetDateTime) {
-                    sb.append("[date " + 
-                            ParsingUtilities.dateToString((OffsetDateTime) v) +"]");
+                    sb.append("[date " +
+                            ParsingUtilities.dateToString((OffsetDateTime) v) + "]");
                 } else if (v instanceof String) {
                     if (quote) {
                         try {
-							sb.append(ParsingUtilities.mapper.writeValueAsString(((String) v)));
-						} catch (JsonProcessingException e) {
-							// will not happen
-						}
+                            sb.append(ParsingUtilities.mapper.writeValueAsString(((String) v)));
+                        } catch (JsonProcessingException e) {
+                            // will not happen
+                        }
                     } else {
                         sb.append((String) v);
                     }
                 } else if (v instanceof Double || v instanceof Float) {
                     Number n = (Number) v;
+                    //CS427 Issue Link: https://github.com/OpenRefine/OpenRefine/issues/3102
+                    final double minNonSciNotation = 0.000_001;
+                    final double maxNonSciNotation = 1_000_000_000_000_000_000_000.0;
+                    final DecimalFormat decimalFormat = new DecimalFormat("0"); //, DecimalFormatSymbols.getInstance(Locale.ENGLISH));
+                    decimalFormat.setMaximumFractionDigits(15);
+                    decimalFormat.setMaximumIntegerDigits(22);
```

Comment:
Those constants are hard-coded here, but if the goal is to match the rendering provided in the data table itself, would it not make sense to have this logic stored only in one place instead? That would avoid those renderings getting out of sync? I have not checked if that makes sense though :)

Reply:
@wetneb Yes I think that definitely makes sense. Do you know where I can access those values in the code without hardcoding them? Thanks!

Reply:
I do not know it on top of my head, perhaps you could find the relevant place by debugging? I suspect it could happen in the frontend, actually. If that is the case, then we cannot refer directly to it from the Java code. We have multiple options:
* shift the rendering of previewed values to the frontend too, so that we can unify rendering there
* introduce a Java class dedicated to value rendering, which replicates what the frontend does, and that can be used outside of the `PreviewExpressionCommand`